### ISSUE TITLE
Unit test failure: io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogTests$SecureBookKeeperLogTests.testReconcileLedgersBadCandidates

### ISSUE THREAD

**Issue by RaulGracia** [pravega/pravega#5515]
**Problem description**
Found this unit test failing on Git Actions: 
```
io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogTests$SecureBookKeeperLogTests > testReconcileLedgersBadCandidates FAILED
    io.pravega.segmentstore.storage.DataLogNotAvailableException: Unable to establish connection to ZooKeeper or BookKeeper.
        at io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory.initialize(BookKeeperLogFactory.java:116)
        at io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogTests.setUp(BookKeeperLogTests.java:170)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.lang.Thread.run(Thread.java:834)
        Caused by:
        java.io.IOException: Failed to initialize metadata client driver
            at org.apache.bookkeeper.client.BookKeeper.<init>(BookKeeper.java:455)
            at org.apache.bookkeeper.client.BookKeeper$Builder.build(BookKeeper.java:298)
            at org.apache.bookkeeper.client.impl.BookKeeperBuilderImpl.build(BookKeeperBuilderImpl.java:86)
            at io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory.startBookKeeperClient(BookKeeperLogFactory.java:189)
            at io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory.initialize(BookKeeperLogFactory.java:104)
            ... 14 more
            Caused by:
            org.apache.bookkeeper.meta.exceptions.MetadataException: Failed to create zookeeper client to localhost:52196
                at org.apache.bookkeeper.meta.zk.ZKMetadataDriverBase.initialize(ZKMetadataDriverBase.java:217)
                at org.apache.bookkeeper.meta.zk.ZKMetadataClientDriver.initialize(ZKMetadataClientDriver.java:62)
                at org.apache.bookkeeper.client.BookKeeper.<init>(BookKeeper.java:445)
                ... 18 more
                Caused by:
                org.apache.zookeeper.KeeperException$ConnectionLossException: KeeperErrorCode = ConnectionLoss
                    at org.apache.zookeeper.KeeperException.create(KeeperException.java:102)
                    at org.apache.bookkeeper.zookeeper.ZooKeeperWatcherBase.waitForConnection(ZooKeeperWatcherBase.java:159)
                    at org.apache.bookkeeper.zookeeper.ZooKeeperClient$Builder.build(ZooKeeperClient.java:257)
                    at org.apache.bookkeeper.meta.zk.ZKMetadataDriverBase.initialize(ZKMetadataDriverBase.java:197)
                    ... 20 more
```

**Problem location**
`BookKeeperLogTests`

**Suggestions for an improvement**
Make the test to pass reliably.

**Comment by andreipaduroiu** [comment]
This is in the same boat as the one where we fail to bind to an address/port.

**Comment by RaulGracia** [comment]
Fixed via #4999.
